e2e_test/schema_registry/alter_sr.slt

```diff
@@ -16,59 +16,68 @@ FORMAT PLAIN ENCODE PROTOBUF(
 statement ok
 CREATE MATERIALIZED VIEW mv_user AS SELECT * FROM src_user;
 
-# Changing type is not allowed
-statement error Feature is not yet implemented: this altering statement will drop columns, which is not supported yet: \(city: character varying\)
-ALTER SOURCE src_user FORMAT PLAIN ENCODE PROTOBUF(
-    schema.registry = 'http://message_queue:8081',
-    message = 'test.UserWithNewType'
-);
-
-# Changing format/encode is not allowed
-statement error Feature is not yet implemented: the original definition is FORMAT Plain ENCODE Protobuf, and altering them is not supported yet
-ALTER SOURCE src_user FORMAT NATIVE ENCODE PROTOBUF(
+statement ok
+CREATE TABLE t_user WITH (
+    connector = 'kafka',
+    topic = 'sr_pb_test',
+    properties.bootstrap.server = 'message_queue:29092',
+    scan.startup.mode = 'earliest'
+)
+FORMAT PLAIN ENCODE PROTOBUF(
     schema.registry = 'http://message_queue:8081',
     message = 'test.User'
 );
 
-statement ok
-ALTER SOURCE src_user FORMAT PLAIN ENCODE PROTOBUF(
-    schema.registry = 'http://message_queue:8081',
-    message = 'test.UserWithMoreFields'
-);
+statement error
+SELECT age FROM mv_user;
 
-# Dropping columns is not allowed
-statement error Feature is not yet implemented: this altering statement will drop columns, which is not supported yet: \(age: integer\)
+statement error
+SELECT age FROM t_user;
+
+# Push more events with extended fields
+system ok
+python3 e2e_test/schema_registry/pb.py "message_queue:29092" "http://message_queue:8081" "sr_pb_test" 5 user_with_more_fields
+
+sleep 5s
+
+# Refresh source schema
+statement ok
 ALTER SOURCE src_user FORMAT PLAIN ENCODE PROTOBUF(
     schema.registry = 'http://message_queue:8081',
     message = 'test.User'
 );
```

Comment:
IIUC, altering table with same format & encoding but different properties is also supported in this PR, right? If so, please update the PR description

```
ALTER [SOURCE|TABLE] <name> FORMAT ... ENCODE ... (
    (... new properties ...)
);
```

Reply:
No, I reverted the implementation of `ALTER TABLE FORMAT ENCODE` as we discussed in slack that it should be suspended and needs more consideration. This pr is only about `ALTER TABLE REFRESH SCHEMA`.

Reply:
I see. Only `ALTER SOURCE` works, `ALTER TABLE ... FORMAT ... ENCODE` is not in this PR.

Reply:
Let me summarize. So after this PR gets merged, it will be like

- [ ] `ALTER TABLE <name> FORMAT ... ENCODE ... ( ... )`
- [x] `ALTER SOURCE <name> FORMAT ... ENCODE ... ( ... )`
- [x] `ALTER TABLE <name> REFRESH SCHEMA`
- [ ] `ALTER SOURCE <name> REFRESH SCHEMA`

Right? Shall we complete the rest two in the future?